File: 274. H-Index/Solution.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        if (( citations.size() == 0 ) || ( *max_element(citations.begin(), citations.end()) == 0 )) { return 0; }
        
        sort(citations.begin(), citations.end(), greater<int>());
        int i = 0;
        while ( i < citations.size() ) {
            if ( citations[i] < i+1 ) { return i; }
            i ++;
        }
        return i;
    }
};

File: 274. H-Index/Solution.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        if len(citations) == 0 or max(citations) == 0:
            return 0
        citations = sorted(citations, reverse=True)
        for idx, v in enumerate(citations):
            if (idx+1 == v) or (idx < len(citations) - 1 and v > idx+1 and idx+2 > citations[idx+1]):
                return idx + 1
        return idx + 1

